src/app/components/iot-gateway/iot-gateway.component.ts

```diff
@@ -339,10 +339,14 @@ export class IotGatewayComponent implements OnInit, AfterViewInit {
   }
 
   generateColor(gatewayName: string) {
-    let hash = 0;
-    for (var i = 0; i < gatewayName.length; i++) {
-      hash = gatewayName.charCodeAt(i) + ((hash << 5) - hash);
+    if (gatewayName == "HelloWorldGroup") {
```

Comment:
Can gatewayName be undefined, or length ==0 ? We should maybe return a default if that's the case.

Reply:
This can never happen, as it is a required field.

Reply:
Looks good then!! Thanks!